drivers/display/display_nrf_led_matrix.c

```diff
@@ -0,0 +1,448 @@
+/*
+ * Copyright (c) 2021, Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <drivers/display.h>
+#include <devicetree.h>
+#include <dt-bindings/gpio/gpio.h>
+#include <hal/nrf_timer.h>
+#ifdef PWM_PRESENT
+#include <hal/nrf_pwm.h>
+#endif
+#include <nrfx_gpiote.h>
+#ifdef PPI_PRESENT
+#include <nrfx_ppi.h>
+#endif
+
+#define MATRIX_NODE  DT_INST(0, nrf_led_matrix)
+#define TIMER_NODE   DT_PHANDLE(MATRIX_NODE, timer)
+#define USE_PWM      DT_NODE_HAS_PROP(MATRIX_NODE, pwm)
+#define ROW_COUNT    DT_PROP_LEN(MATRIX_NODE, row_gpios)
+#define COL_COUNT    DT_PROP_LEN(MATRIX_NODE, col_gpios)
+
+#define X_PIXELS     DT_PROP(MATRIX_NODE, width)
+#define Y_PIXELS     DT_PROP(MATRIX_NODE, height)
+#define PIXEL_COUNT  DT_PROP_LEN(MATRIX_NODE, pixel_mapping)
+BUILD_ASSERT(PIXEL_COUNT == (X_PIXELS * Y_PIXELS),
+	     "Invalid length of pixel-mapping");
+
+#define PIXEL_MAPPING(idx)  DT_PROP_BY_IDX(MATRIX_NODE, pixel_mapping, idx)
+#define CHECK_PIXEL(node_id, pha, idx) \
+	BUILD_ASSERT((PIXEL_MAPPING(idx) >> 4) < ROW_COUNT, \
+		     "Invalid row index in pixel-mapping["#idx"]"); \
+	BUILD_ASSERT((PIXEL_MAPPING(idx) & 0xF) < COL_COUNT, \
+		     "Invalid column index in pixel-mapping["#idx"]");
+DT_FOREACH_PROP_ELEM(MATRIX_NODE, pixel_mapping, CHECK_PIXEL)
+
+#define BASE_FREQUENCY     16000000
+#define TIMER_CLK_CONFIG   NRF_TIMER_FREQ_16MHz
+#define PWM_CLK_CONFIG     NRF_PWM_CLK_16MHz
+#define REFRESH_FREQUENCY  50
+#define BRIGHTNESS_MAX     255
```

Comment:
some of these parameters, e.g. `REFRESH_FREQUENCY` could be defined in DT.

Reply:
Added the `refresh-frequency` DT property.